core/src/main/java/hudson/slaves/NodeProvisioner.java

```diff
@@ -214,7 +210,8 @@ public void run() {
 
                     int plannedCapacitySnapshot = 0;
 
-                    for (Iterator<PlannedNode> itr = pendingLaunches.iterator(); itr.hasNext(); ) {
+                    List<PlannedNode> snapPendingLaunches = new ArrayList<PlannedNode>(pendingLaunches.get());
```

Comment:
NIT: maybe call `getPendingLaunches`. Seeing as you have it, you may as well use it.